ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,94 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
             sub_faces.extend(face.sub_faces_by_ratio(ratio))
         return sub_faces
 
+    def _sub_faces_by_dimensions(self, sub_face_height, sub_face_width, wall_length,
+                                dist_breakup, sill_height, rect_height, changed_var):
+        """Get a list of faces with a combined area equal to the glazing ratio
+        times this face area. The glazing ratio is derived internally using the
+        specified sub face height and width along with other parameters.
+
+        All sub faces will lie inside the boundaries of this face.
+
+        Args:
+            height: sub face height
+            width: sub face width
+            wall_length: total length (width) of sub face wall
+            dist_breakup: distance by which main rectangle is meant to be 
+                divided up into points on the bottom
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. Note that, if the
+                ratio is too large for the height, the ratio will take precedence
+                and the sub-rectangle height will be smaller than this value.
+            rect_height: A number for the target height of the output rectangles. Note that, if the ratio is too large for the height,
+                the ratio will take precedence and the sub-rectangle height will
+                be larger than this value.
+            changed_var: string indicating the variable being changed 
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        sub_face_height_final = rect_height if sub_face_height > rect_height else sub_face_height
```

Comment:
I think we will need a tolerance factor in here. The original Javascript wasn't producing geometry for EnergyPlus and so we didn't have to worry about windows going all of the way to the edge of of the parent face but this is illegal in EnergyPlus. So I think this would have to be:
```
sub_face_height_final = rect_height - 0.01 if sub_face_height > rect_height - 0.01 else sub_face_height